agent/config/config.go

```diff
@@ -297,6 +304,25 @@ func environmentConfig() Config {
 		seelog.Warnf("Invalid format for \"ECS_NUM_IMAGES_DELETE_PER_CYCLE\", expected an integer. err %v", err)
 	}
 
+	instanceAttributesEnv := os.Getenv("ECS_INSTANCE_ATTRIBUTES")
+	attributeDecoder := json.NewDecoder(strings.NewReader(instanceAttributesEnv))
+	var instanceAttributes map[string]string
+
+	err = attributeDecoder.Decode(&instanceAttributes)
+	if err != io.EOF && err != nil {
+		msg := "Invalid format for ECS_INSTANCE_ATTRIBUTES. Expected a json hash"
```

Comment:
I tested [this](https://play.golang.org/p/3rZHg5fAJW) and it looks like you do get an error back from `json.Unmarshal` (used by `decoder.Decode`).  So really all that needs to happen here is updating the error message to be more descriptive.